src/name.rs

```diff
@@ -92,6 +92,18 @@ impl std::borrow::Borrow<str> for Name {
     }
 }
 
+// NOTE: This custom function ensures that serde serializes to a `Token::Str`
+// rather than `Token::NewType` that appears with a derived `Serialize`.
+impl Serialize for Name {
```

Comment:
can we use [#[serde(transparent)]](https://serde.rs/container-attrs.html#transparent)?